centreon/src/Core/HostCategory/Application/UseCase/FindHostCategory/FindHostCategory.php

```diff
@@ -0,0 +1,147 @@
+
+final class FindHostCategory
+{
+    use LoggerTrait;
+
+    /**
+     * @param ReadHostCategoryRepositoryInterface $readHostCategoryRepository
+     * @param ReadAccessGroupRepositoryInterface $readAccessGroupRepositoryInterface
+     * @param ContactInterface $user
+     */
+    public function __construct(
+        private ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private ReadAccessGroupRepositoryInterface $readAccessGroupRepositoryInterface,
+        private ContactInterface $user
+    ) {
+    }
+
+    /**
+     * @param int $hostCategoryId
+     * @param PresenterInterface $presenter
+     */
+    public function __invoke(int $hostCategoryId, PresenterInterface $presenter): void
+    {
+        try {
+            if ($this->user->isAdmin()) {
+                if (! $this->readHostCategoryRepository->exists($hostCategoryId)) {
+                    $this->error('Host category not found', [
+                        'hostcategory_id' => $hostCategoryId,
+                    ]);
+                    $presenter->setResponseStatus(new NotFoundResponse('Host category'));
+
+                    return;
+                }
+
+                $this->retrieveObjectAndSetResponse($presenter, $hostCategoryId);
+            } elseif (
+                $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ)
+                || $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)
+            ) {
+                $this->debug(
+                    'User is not admin, use ACLs to retrieve a host category',
+                    ['user_id' => $this->user->getId()]
+                );
+                $accessGroups = $this->readAccessGroupRepositoryInterface->findByContact($this->user);
+
+                if (! $this->readHostCategoryRepository->existsByAccessGroups($hostCategoryId, $accessGroups)) {
+                    $this->error('Host category not found', [
+                        'hostcategory_id' => $hostCategoryId,
+                        'accessgroups' => $accessGroups,
+                    ]);
+                    $presenter->setResponseStatus(new NotFoundResponse('Host category'));
+
+                    return;
+                }
+
+                $this->retrieveObjectAndSetResponse($presenter, $hostCategoryId);
+            } else {
+                $this->error('User doesn\'t have sufficient rights to see a host category', [
+                    'user_id' => $this->user->getId(),
+                    'host_category_id' => $hostCategoryId,
+                ]);
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostCategoryException::accessNotAllowed()->getMessage())
+                );
+            }
+        } catch (\Throwable $ex) {
+            $presenter->setResponseStatus(
+                new ErrorResponse(HostCategoryException::findHostCategory($ex, $hostCategoryId)->getMessage())
```

Comment:
We should send the object in order to avoid a double-translation.
```suggestion
                new ErrorResponse(HostCategoryException::findHostCategory($ex, $hostCategoryId))
```